src/app/components/client/BillingPeriod.tsx

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ReactNode, createContext, useContext, useRef } from "react";
+import {
+  RadioGroupProps,
+  RadioGroupState,
+  useRadioGroupState,
+} from "react-stately";
+import {
+  AriaRadioGroupProps,
+  AriaRadioProps,
+  mergeProps,
+  useFocusRing,
+  useRadio,
+  useRadioGroup,
+} from "react-aria";
+import styles from "./BillingPeriod.module.scss";
+import { useL10n } from "../../hooks/l10n";
+import { VisuallyHidden } from "../server/VisuallyHidden";
+
+export type BillingPeriod = "yearly" | "monthly";
+
+export type Props = {
+  defaultValue?: BillingPeriod;
+  onChange: (_selectedBillingPeriod: BillingPeriod) => void;
+};
+
+export const BillingPeriodToggle = (props: Props) => {
+  const l10n = useL10n();
+
+  return (
+    <>
```

Comment:
Does this need the wrapper fragment since there is only one child, `<RadioGroup>`?

Reply:
It does not: 44f3d0557